src/main/java/com/google/songgame/servlets/ChatServlet.java

```diff
@@ -44,6 +50,42 @@ private Map readJSONFromRequest(HttpServletRequest request) throws IOException {
     return jsonData;
   }
 
+  private Map createPusherChatResponse(Map data) {
```

Comment:
+1. Since it may not be obvious how to parse the JSON into a specific Map type, here's an example:
https://stackoverflow.com/q/2779251

Let us know if you run into trouble with this and we can help you, since working with generics in Java can be tricky. But like Aleksey said, this will make it so you don't have to do any type casting like you do on lines 55/56